contribs/gmf/src/directives/timeslider.js

```diff
@@ -0,0 +1,182 @@
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfTimeSlider
+ */
+gmf.TimeSlider = function(gmfTimeSliderTemplateUrl, $timeout, $filter) {
+  return {
+    scope : {
+      onDateSelected : '&',
+      time : '='
+    },
+    bindToController : true,
+    controller : 'gmfTimeSliderController',
+    controllerAs : 'sliderCtrl',
+    restrict: 'AE',
+    templateUrl : gmfTimeSliderTemplateUrl,
+    link: function(scope, element, attrs, ctrl) {
+
+      /** @type {string} */
+      var dateFormat = 'shortDate';
+
+      ctrl.sliderOptions['slide'] = updateTooltip_;
+      ctrl.sliderOptions['start'] = updateTooltip_;
+      ctrl.sliderOptions['stop'] = onSliderReleased_;
+      $timeout(init_);
+
+      function init_() {
+        if (element.children().first()[0].offsetHeight) {
+          //element is not hidden, init tooltip is allowed
+          element.find('span.ui-slider-handle').each(initToolTip_);
+        } else {
+          //element is hidden. Do not init tooltip now
+          $('body').on('shown.bs.popover', function(jqEvt) {
+            //Element contained in a popover ?
+            if ($('.popover[role="tooltip"]').find(element).length) {
+              element.find('span.ui-slider-handle').each(initToolTip_);
+            }
+          });
+        }
+
+        element.find('.ui-slider-handle').on('shown.bs.tooltip', function() {
+          //Display sibling (only for range time).
+          var otherHandle = $(this).siblings('.ui-slider-handle');
+          if (otherHandle.length > 0) {
+            //test if other handle has its toolip hidden (do not call show for nothing)
+            if (!$(otherHandle[0]).attr('aria-describedby')) {
+              otherHandle.tooltip('show');
+            }
+          }
+        });
+
+        element.find('.ui-slider-handle').on('hide.bs.tooltip', function(jqEvt) {
+          if (element.find(this).length > 0) {
+            jqEvt.preventDefault();
+          }
+        });
+      }
+
+      function updateTooltip_(e, slider) {
+        var handle = $(slider.handle);
+        var left  = handle.position().left;
+        var tooltip = $('#' + handle.attr('aria-describedby'));
+        var dateText  = $filter('date')(new Date(slider.value), dateFormat);
+        handle.attr('data-original-title', dateText).tooltip('fixTitle');
+        tooltip.css('left', left - tooltip.width() / 2);
+        tooltip.find('.tooltip-inner').text(dateText);
+      }
+
+      function initToolTip_(i) {
+        var el = $(this);
+        if (!el.attr('aria-describedby')) {
+          var dateText  = Array.isArray(ctrl.dates) ?
+          $filter('date')(new Date(ctrl.dates[i]), dateFormat) :
+          $filter('date')(new Date(ctrl.dates), dateFormat);
+
+          el.tooltip({
+            'selector': 'span.ui-slider-handle',
+            'trigger': 'manual',
+            'placement' : i === 0 && 'bottom' || 'top',
```

Comment:
Also, though I think I understand why you put one at the bottom and one at the top. This makes things really weird at first. I'd prefer showing the tooltips only on hover.